src/Hub.cpp

```diff
@@ -271,6 +274,16 @@ sc2::Tag Hub::AssignBuildTask(Order* order_, const sc2::Point2D& point_) {
     return worker->Tag();
 }
 
+bool Hub::AssignRepairTask(const sc2::Unit& damaged_) {
+    const sc2::Point2D repair_location(damaged_.pos);
+    Worker* worker = GetClosestFreeWorker(repair_location);
```

Comment:
'GetClosestFreeWorker' marks the worker as 'busy' before it returns the worker.

The reason I have it in Hub, is so it can be called by Strategy to micro Combat Repair of Mechanical Units. We don't yet have a way for Strategy to talk to another Plugin. Even a 'Strategy Settings' struct would not allow Strategy to dynamically call RepairMan::AssignRepairTask for damaged military units. I really do think Hub should have a pointer to every other plugin, like CommandCenter does.

Hub contains the free/busy_workers list, so I feel that Hub should manage the actual job assignment just like 'AssignRefineryConstruction', 'AssignBuildTask', 'AssignVespeneHarvester'. The Repairman contains the logic on when it should happen.

Reply:
According to my design Hub is a cache. It contains a picture of the real world with all these lists and additional info the bit needs. All the game logic should be moved out of it and the hub itself should be simplified a lot (e.g. all that different lists should gone).
The subsystems act independently until a special messaging channel is set, such approach allows to make very agile system with clear division between parts. Also, the subsystems divided by purpose intentionally, it allows to keep related code inside. The idea about single point of taking decision and executing them immediately could work but not in this architecture.

So, I am don't like the idea of creating a command center out of Hub.

Reply:
Okay, moved AssignRepairTask to Repairman.